witness/src/Dscp/Snowdrop/Expanders.hs

```diff
@@ -45,92 +47,115 @@ expandGTxs ::
        , HasLens ctx RestrictCtx
        , HasLens ctx (ChgAccumCtx ctx)
        )
-    => [GTxWitnessed]
+    => FeeCoefficients
+    -> [GTxWitnessed]
     -> ERoComp Exceptions Ids Values ctx [StateTx Ids Proofs Values]
-expandGTxs txs = expandUnionRawTxs getByGTx txs
+expandGTxs coeffs txs
+    = expandUnionRawTxs getByGTx
+    $ map (turnIntoFees $ calculateFee coeffs) txs
 
--- | Expand list of global txs.
+-- | Expand one global tx.
 expandGTx ::
        ( Default (ChgAccum ctx)
        , HasLens ctx RestrictCtx
        , HasLens ctx (ChgAccumCtx ctx)
        )
-    => GTxWitnessed
+    => FeeCoefficients
```

Comment:
What's the hassle here? Why do you pass `FeeCoefficients` directly instead of using reflection? It is obviously inconvenient. Don't be afraid to pull the `WitnessContext` altogether, it was designed exactly for this type of usage.

Reply:
This ^